Details of declarativeNetPequest permission notifications (#38667)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/declarativenetrequest/index.md

## Permissions

> To use this API, an extension must request the `"declarativeNetRequest"` or `"declarativeNetRequestWithHostAccess"` [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) in its [`manifest.json`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json) file. The `"declarativeNetRequest"` is notified to users during initial installation and, when added to an installed extension, updates. `"declarativeNetRequestWithHostAccess"` isn't notified to users, as it relies on host permissions.

Reviewer: "xxx permission is/isn't notified to users" looks rather passive and unspecific. I'm not sure if a developer would make the connection to the concept of "permission warning" (which is the terminology that Chrome uses and appears in some extension APIs). How about this one:

```suggestion
To use this API, an extension must request the `"declarativeNetRequest"` or `"declarativeNetRequestWithHostAccess"` [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) in its [`manifest.json`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json) file. The `"declarativeNetRequest"` permission is shown to users in permission prompts, the `"declarativeNetRequestWithHostAccess"` is not.
```

(I dropped the host permissions bit again because the next paragraph clarifies that even further.)

> To use this API, an extension must request the `"declarativeNetRequest"` or `"declarativeNetRequestWithHostAccess"` [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) in its [`manifest.json`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json) file. The `"declarativeNetRequest"` is notified to the user during installation, while `"declarativeNetRequestWithHostAccess"` is granted without notification. For more information, see [Request the right permissions](https://extensionworkshop.com/documentation/develop/request-the-right-permissions/) on Extension Workshop.

Reviewer: ```suggestion
To use this API, an extension must request the `"declarativeNetRequest"` or `"declarativeNetRequestWithHostAccess"` [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) in its [`manifest.json`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json) file. The `"declarativeNetRequest"` is notified to the user during installation (and updates, if newly added), while `"declarativeNetRequestWithHostAccess"` does not have a separate notification, as it relies on host permissions.
```

I dropped the pointer to the Extension Workshop because this behavior is cross-browser, and instead opted for writing the relevant differences directly here. If you think that there's something unclear, or something uniquely useful in the extension workshop, let me know.

Author: ```suggestion
To use this API, an extension must request the `"declarativeNetRequest"` or `"declarativeNetRequestWithHostAccess"` [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) in its [`manifest.json`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json) file. The `"declarativeNetRequest"` is notified to users during initial installation and, when added to an installed extension, updates. `"declarativeNetRequestWithHostAccess"` isn't notified to users, as it relies on host permissions.
```
